modules/git/url/url.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package url
+
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+// URL represents a git remote URL
+type URL struct {
+	*url.URL
+	extraMark int // 0 no extra 1 scp 2 file path with no prefix
```

Comment:
Could we use `URL.Schema` directly without introducing `extraMark`?
* URL.Schema="": file
* URL.Schema="ssh":  user@host:path
* URL.Schema="http"/"https": ...

Reply:
We can't, we need to know the difference between `file:///root` and `/root`.

Reply:
What's the difference?